diff --git a/src/dawn/native/CMakeLists.txt b/src/dawn/native/CMakeLists.txt
index 147fe8ad90..c68d650f20 100644
--- a/src/dawn/native/CMakeLists.txt
+++ b/src/dawn/native/CMakeLists.txt
@@ -1105,16 +1105,18 @@ if (DAWN_BUILD_MONOLITHIC_LIBRARY)
             target_include_directories(${target_name} PRIVATE
                 "${DAWN_SWITCH_NVK_ROOT}/include")
             target_link_libraries(${target_name} PRIVATE
+                -Wl,--start-group
                 webgpu_dawn
                 "${DAWN_SWITCH_NVK_LIBRARY}"
-                drm_nouveau
-                zstd
-                z
+                ${DAWN_SWITCH_EXTRA_LIBRARIES}
                 nx
-                m)
+                m
+                -Wl,--end-group)
             target_link_options(${target_name} PRIVATE
                 -Wl,--gc-sections
-                -Wl,--allow-multiple-definition)
+                -Wl,--allow-multiple-definition
+                -Wl,-u,vk_icdGetInstanceProcAddr
+                -Wl,-u,vk_icdNegotiateLoaderICDInterfaceVersion)
 
             set(_nacp "${CMAKE_CURRENT_BINARY_DIR}/${target_name}.nacp")
             set(_nro "${CMAKE_CURRENT_BINARY_DIR}/${target_name}.nro")
